sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -1210,6 +1210,38 @@ class DataSourceV2SQLSuite
     }
   }
 
+  test("LOAD DATA INTO TABLE") {
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      sql(
+        s"""
+           |CREATE TABLE $t (id bigint, data string)
+           |USING foo
+           |PARTITIONED BY (id)
+         """.stripMargin)
+
+      val e1 = intercept[AnalysisException] {
```

Comment:
Refactor this a bit?
```Scala
def validate(sqlStr: String): Unit = {
  val e = intercept[AnalysisException] {
    sql(sqlStr)
  }
  assert(e.message.contains("LOAD DATA is only supported with v1 tables"))
}

validate(s"LOAD DATA INPATH 'filepath' INTO TABLE $t")
validate(s"LOAD DATA LOCAL INPATH 'filepath' INTO TABLE $t")
...
```

Or how about creating a function that [SPARK-29481](https://issues.apache.org/jira/browse/SPARK-29481) can reuse?

Reply:
I actually created the function that can be reused here: #26183. Please let me know what you think. Thanks!

Reply:
It looks good. Then I won't create another here, I can rebase this after #26183 is merged.